File: crates/codeprysm-core/tests/fixtures/integration_repos/python/sample.py
"""Sample Python module for integration testing.

This module demonstrates various Python language features for
graph generation validation.
"""

from typing import Optional, List
import asyncio


# Module-level constant
MAX_ITEMS = 100


def standalone_function(param: str) -> str:
    """A standalone function outside any class."""
    return f"processed_{param}"


async def async_standalone(url: str) -> dict:
    """An async standalone function."""
    await asyncio.sleep(0.1)
    return {"url": url}


class Calculator:
    """A simple calculator class with methods and fields."""

    class_constant = 3.14159

    def __init__(self, initial_value: int = 0):
        """Initialize the calculator with an initial value."""
        self.value = initial_value
        self.history: List[int] = []

    def add(self, amount: int) -> int:
        """Add an amount to the current value."""
        self.value += amount
        self.history.append(amount)
        return self.value

    def multiply(self, factor: int) -> int:
        """Multiply the current value by a factor."""
        self.value *= factor
        return self.value

    @staticmethod
    def square(x: int) -> int:
        """Static method to square a number."""
        return x * x

    @classmethod
    def from_string(cls, value_str: str) -> "Calculator":
        """Class method to create from string."""
        return cls(int(value_str))


class AsyncProcessor:
    """A class with async methods."""

    def __init__(self, name: str):
        self.name = name
        self.processed_count = 0

    async def process_item(self, item: str) -> str:
        """Async method to process an item."""
        await asyncio.sleep(0.01)
        self.processed_count += 1
        return f"{self.name}:{item}"

    async def process_batch(self, items: List[str]) -> List[str]:
        """Async method to process multiple items."""
        results = []
        for item in items:
            result = await self.process_item(item)
            results.append(result)
        return results


def decorator_example(func):
    """A simple decorator function."""
    def wrapper(*args, **kwargs):
        return func(*args, **kwargs)
    return wrapper


@decorator_example
def decorated_function(x: int) -> int:
    """A function with a decorator."""
    return x + 1


class InheritedClass(Calculator):
    """A class that inherits from Calculator."""

    def __init__(self, initial_value: int = 0, precision: int = 2):
        super().__init__(initial_value)
        self.precision = precision

    def divide(self, divisor: int) -> float:
        """Divide the current value."""
        if divisor == 0:
            raise ValueError("Cannot divide by zero")
        result = self.value / divisor
        self.value = int(result)
        return round(result, self.precision)

File: crates/codeprysm-core/tests/fixtures/integration_repos/c/sample.c
/**
 * Sample C module for integration testing.
 *
 * This module demonstrates various C language features for
 * graph generation validation.
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

/* Module-level constant */
#define MAX_ITEMS 100

/* Type definitions */
typedef int ErrorCode;
typedef char* String;

/* Enum definition */
typedef enum {
    STATUS_OK = 0,
    STATUS_ERROR = 1,
    STATUS_PENDING = 2
} Status;

/* Forward declarations */
struct Calculator;
typedef struct Calculator Calculator;

/* Struct definition with fields */
struct Calculator {
    int value;
    int history[MAX_ITEMS];
    int history_count;
};

/* Another struct definition */
typedef struct {
    char* name;
    int processed_count;
} Processor;

/* Union definition */
typedef union {
    int int_value;
    float float_value;
    char* string_value;
} Value;

/* Function to create a new calculator */
Calculator* calculator_new(int initial_value) {
    Calculator* calc = (Calculator*)malloc(sizeof(Calculator));
    if (calc) {
        calc->value = initial_value;
        calc->history_count = 0;
    }
    return calc;
}

/* Function to free a calculator */
void calculator_free(Calculator* calc) {
    if (calc) {
        free(calc);
    }
}

/* Function to add a value */
int calculator_add(Calculator* calc, int amount) {
    if (!calc) return -1;

    calc->value += amount;
    if (calc->history_count < MAX_ITEMS) {
        calc->history[calc->history_count++] = amount;
    }
    return calc->value;
}

/* Function to multiply by a factor */
int calculator_multiply(Calculator* calc, int factor) {
    if (!calc) return -1;

    calc->value *= factor;
    return calc->value;
}

/* Function to get the current value */
int calculator_get_value(const Calculator* calc) {
    return calc ? calc->value : 0;
}

/* Standalone function outside struct context */
int standalone_function(const char* param) {
    return param ? (int)strlen(param) : 0;
}

/* Static helper function */
static int square(int x) {
    return x * x;
}

/* Function with pointer parameter */
void process_array(int* arr, int size) {
    for (int i = 0; i < size; i++) {
        arr[i] = square(arr[i]);
    }
}

/* Function returning struct pointer */
Processor* processor_new(const char* name) {
    Processor* proc = (Processor*)malloc(sizeof(Processor));
    if (proc) {
        proc->name = strdup(name);
        proc->processed_count = 0;
    }
    return proc;
}

/* Function with multiple parameters */
ErrorCode process_item(Processor* proc, const char* item, int* result) {
    if (!proc || !item || !result) {
        return STATUS_ERROR;
    }

    proc->processed_count++;
    *result = (int)strlen(item);
    return STATUS_OK;
}

/* Main function for testing */
int main(int argc, char** argv) {
    Calculator* calc = calculator_new(0);
    calculator_add(calc, 10);
    calculator_multiply(calc, 2);

    printf("Result: %d\n", calculator_get_value(calc));

    calculator_free(calc);
    return 0;
}
